### PR TITLE
Refactor & improvements

### PR DESCRIPTION
**Pull request by fgheysels**
- rename AsyncMimeParser to MimeParser
- check content-type before parsing to MIME message
- read stream content async
- async decompression
- prevent possible nullreferenceexception

### CODE REVIEW

File: src/extension/MimeInspector/Utilities/MimeParser.cs [hunk 1]
```diff
@@ -43,17 +43,44 @@ public static MimeMessage ParseMessage(Stream inputStream, HTTPHeaders headers,
...
 
-            using (ChainedStream streamWithHeaders = new ChainedStream())
+            if(IsMimeMessage(headerArray) == false)
             {
-                streamWithHeaders.Add(new MemoryStream(Encoding.UTF8.GetBytes(headerString)), false);
-                streamWithHeaders.Add(inputStream, false);
+                return null;
+            }
+
+            string headerString = String.Join("\r\n", headerArray.Select(h => $"{h.Name}:{h.Value}"));
```
Comment (samneirinck) [inline src/extension/MimeInspector/Utilities/MimeParser.cs]: Should there be a space between header name and value?

**Review by samneirinck — CHANGES_REQUESTED** [review]

File: src/extension/MimeInspector/Utilities/MimeParser.cs (hunk 1 above)
Comment (fgheysels) [reply src/extension/MimeInspector/Utilities/MimeParser.cs]: [RFC 7230](https://tools.ietf.org/html/rfc7230#section-3.2) states that the whitespace is optional.  But if you rather have a whitespace, I'll add one.  Just let me know

Comment (samneirinck) [reply src/extension/MimeInspector/Utilities/MimeParser.cs]: Today I learned something new :). If it's not disallowed I would add it for readability

Comment (fgheysels) [reply src/extension/MimeInspector/Utilities/MimeParser.cs]: done.

**Review by samneirinck — APPROVED** [review]

**Comment by samneirinck** [comment]
Thank you, it's in https://github.com/samneirinck/MimeInspector/releases/tag/v0.3.0

**Pull request merged**